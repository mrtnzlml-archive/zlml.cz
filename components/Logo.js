// @flow

import * as React from 'react';

import Colors from '../services/Colors';
import Link from '../components/markup/Link';
import Paragraph from '../components/markup/Paragraph';

export default () => [
  <Paragraph key="pre-heading">¡Hola! My name is</Paragraph>,
  <h1 key="heading">
    <Link prefetch href="/">
      Martin Zlámal
    </Link>
    <style jsx>{`
      h1 {
        font-size: 5rem;
      }
      @media screen and (min-width: 720px) {
        h1 {
          font-size: 8rem;
        }
      }
      h1 a,
      h1 a:hover,
      h1 a:focus,
      h1 a:visited {
        color: ${Colors.red};
        text-decoration: none;
      }
    `}</style>
  </h1>,
];
