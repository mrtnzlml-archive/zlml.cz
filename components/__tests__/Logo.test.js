// @flow

import * as React from 'react';
import renderer from 'react-test-renderer';
import Router from 'next/router';

import Logo from '../Logo';

// Workaround for "No router instance found. You should only use "next/router" inside the client side of your app."
Router.router = {
  prefetch: () => {},
};

it('renders correctly', () => {
  expect(renderer.create(<Logo />).toJSON()).toMatchSnapshot();
});
