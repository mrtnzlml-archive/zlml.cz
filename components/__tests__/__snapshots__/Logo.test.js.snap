// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <p
    style={
      Object {
        "fontSize": "2rem",
        "hyphens": "auto",
        "textAlign": "justify",
      }
    }
  >
    ¡Hola! My name is
  </p>,
  <h1
    className="jsx-2345896288"
  >
    <span
      className="jsx-3140515621"
    >
      <a
        className="jsx-3140515621"
        href="/"
        onClick={[Function]}
      >
        Martin Zlámal
      </a>
    </span>
  </h1>,
]
`;
